client/lib/site/computed-attributes.js

```diff
@@ -0,0 +1,61 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import trim from 'lodash/trim';
+
+/**
+ * Internal dependencies
+ */
+import { isHttps } from 'lib/url';
+import config from 'config';
+
+export default function( site ) {
+	const attributes = {};
+
+	// If the user has no access to site.options create it as an empty
+	// attribute to avoid potential errors when trying to access its sub properties
+	attributes.options = site.options || {};
+
+	// Add URL without protocol as a `domain` attribute
+	if ( site.URL ) {
+		attributes.domain = site.URL.replace( /^https?:\/\//, '' );
+		attributes.slug = attributes.domain.replace( /\//g, '::' );
+	}
+	attributes.title = trim( site.name ) || attributes.domain;
```

Comment:
Since a fair few tests don't supply `name`, I'd prefer to leave this as it is, to avoid bloating the diff.